crates/holochain/src/core/sys_validate.rs

```diff
@@ -510,150 +506,6 @@ fn check_prev_action_chain<A: ChainItem>(
     }
 }
 
-/// If we are not holding this action then
-/// retrieve it and send it as a RegisterAddLink DhtOp
-/// to our incoming_dht_ops_workflow.
-///
-/// Apply a checks callback to the Record.
-///
-/// Additionally sys validation will be triggered to
-/// run again if we weren't holding it.
-pub async fn check_and_hold_register_add_link<F>(
```

Comment:
These aren't required any more. Doing the `check_and_hold` was redundant as we've already called `retrieve` to get the record back. Mixing the incoming ops send into this step was an unnecessary complication now we have a set of deps in memory that we can read after sys validation has finished it's main task.